tests/functional/metadata/test_update.py

```diff
@@ -0,0 +1,181 @@
+            "POST",
+            url="/api/v1/artifacts",
+            json={"targets": targets},
+        )
+        data_result = result.json()["data"]
+        LOGGER.info(f"Added task_id: {data_result['task_id']}")
+        pytest.rstuf_added_targets.append(result.json())
+        time.sleep(0.5)
+
+    LOGGER.info(
+        "Stop adding artifacts. "
+        f"Total requests: {len(pytest.rstuf_added_targets )}"
+    )
+
+
+@scenario(
+    "../../features/metadata/update.feature",
+    "Updating Root metadata full signed",
+)
+def test_updating_root_metadata_full_signed():
+    """Updating Root metadata full signed."""
+
+
+@given("RSTUF is running and operational")
+def running_multiple_requests():
+    pass
+
+
+@then(
+    "the RSTUF is receiving multiple requests",
+    target_fixture="send_rstuf_requests",
+)
+def rstuf_receiving_requests(http_request):
+    stop_requests = threading.Event()
+    thread = threading.Thread(
+        target=rstuf_requests, args=(stop_requests, http_request)
+    )
+    thread.start()
+    return stop_requests
+
+
+@when(
+    "the RSTUF key holders send a fully signed metadata",
+    target_fixture="response",
+)
+def send_signed_update_metadata(send_rstuf_requests, http_request):
+    # Start thread sending requests to add artifacts
+    pytest.rstuf_thread = send_rstuf_requests
+    # Wait some requests in the broker before run metadata update
+    time.sleep(2)
+    LOGGER.info("[METADATA UPDATE] Submiting Root Metadata Update")
+    with open("metadata-update-payload.json") as f:
+        paylaod_json = json.loads(f.read())
+
+    result = http_request(
+        "POST",
+        url="/api/v1/metadata",
+        json=paylaod_json,
+    )
+    return result
+
+
+@then(
+    "the API requester should get status code '202' with 'task_id'",
+    target_fixture="task_id",
+)
+def task_is_received(response):
+    assert response.status_code == 202, pytest.rstuf_thread.set()
+
+    json_response = response.json()
+    task_id = json_response["data"]["task_id"]
+    LOGGER.info(f"[METADATA UPDATE]  Metadata Updated by {task_id}")
+    return task_id
+
+
+@then(
+    "the API requester gets from endpoint 'GET /api/v1/task' status 'SUCCESS'"
+)
+def task_is_finished(task_id, http_request):
+    count = 0
+
+    # check for the metadata update task
+    while count < 60:
+        response = http_request(
+            "GET",
+            url=f"/api/v1/task/?task_id={task_id}",
+        )
+
+        data = response.json()["data"]
+        state = data.get("state", None)
+        status = data.get("result", {}).get("status")
+        if state == "SUCCESS":
+            LOGGER.info(f"[METADATA UPDATE] {response.text}")
+            assert status is True, pytest.rstuf_thread.set()
+            break
+        else:
+            count += 1
+            time.sleep(0.5)
+
+    LOGGER.info("[METADATA UPDATE] Update Metadata to 2.root.json finished")
+    assert count < 60, pytest.rstuf_thread.set()
+    time.sleep(2)
```

Comment:
The flow I see when you send requests is the following:
1. first you start sending requests before the metadata update
2. requests don't stop while doing a metadata update
3. requests continue 2 seconds after metadata update has finished

Am I correct with the steps?
Do you sleep for 2 seconds, so there could be requests after the metadata update?
If you do that, please add a comment for that, so it's clear. It seems I missed it.

Reply:
Done